components/usage/pkg/db/usage.go

```diff
@@ -207,6 +207,30 @@ type Balance struct {
 	CreditCents   CreditCents   `gorm:"column:creditCents;type:bigint;" json:"creditCents"`
 }
 
+func GetBalance(ctx context.Context, conn *gorm.DB, attributionId AttributionID) (CreditCents, error) {
+	rows, err := conn.WithContext(ctx).
+		Model(&Usage{}).
+		Select("sum(creditCents) as balance").
+		Where("attributionId = ?", string(attributionId)).
+		Group("attributionId").
+		Rows()
+	if err != nil {
+		return 0, fmt.Errorf("failed to get rows for list balance query: %w", err)
+	}
+	defer rows.Close()
+
+	if !rows.Next() {
+		return 0, nil
+	}
+
+	var balance CreditCents
+	err = conn.ScanRows(rows, &balance)
+	if err != nil {
+		return 0, fmt.Errorf("failed to scan row: %w", err)
+	}
+	return balance, nil
```

Comment:
It should be possible to do it this way, if you wanted
```
var balance Balance
	tx := conn.WithContext(ctx).
		Table((&Usage{}).TableName()).
		Select("attributionId, sum(creditCents) as creditCents").
		Where("attributionId = ?", string(attributionId)).
		Group("attributionId").
		First(&balance)
	if tx.Error != nil {
		if tx.Error == gorm.ErrRecordNotFound {
			return 0, nil
		}
		
		return 0, fmt.Errorf("failed to get rows for list balance query: %w", tx.Error)
	}

	return balance.CreditCents, nil
```

Reply:
Why would you prefer that? btw. I realized I unnecessarily added a group by clause. Will remove that and add the suggested test in a follow-up PR.

Reply:
Just because it removes the need for the manual iteration over rows, which can hurt readability for those not familiar with gorm. It also handles the not found case a bit more explicitly which helps readability. But I don't mind, just wanted to offer an alternative in case you wanted to go with it.